src/BuildingRegistry/Building/Building.cs

```diff
@@ -257,37 +257,69 @@ private void GuardActiveBuilding()
             }
         }
 
-        private void AddCommonBuildingUnitIfNeeded(
+        private void AddOrUpdateStatusCommonBuildingUnit(
```

Comment:
This method is getting big. Keep an eye on it during next development in buildingregistry to see if this can be refactored.

Reply:
We were also on the verge of refactoring it. We'll take it along.